### PR TITLE
i173: write middleware unit tests for auth user flows 

### LINKED ISSUE #173: write middleware unit tests for auth user flows

### PR DESCRIPTION
**Pull request by tamari-gray**
fixes #173

### CODE REVIEW

File: lib/middleware/auth/sign_out_user.dart [hunk 1]
```diff
@@ -11,8 +11,7 @@ class SignOutUserMiddleware extends TypedMiddleware<AppState, SignOutUser> {
           // sign out and dispatch the resulting problem if there is one
           final actionAfterSignout = await authService.signOut();
 
-          if (actionAfterSignout != null) {
-            store.dispatch(actionAfterSignout);
-          }
+          // dispatch action to show any errors from signing out
```
Comment (tamari-gray) [inline lib/middleware/auth/sign_out_user.dart:14]: tiny refactor

File: test/unit/middleware/check_platform_middleware_test.dart [hunk 2]
```diff
@@ -1,31 +1,27 @@
 import 'package:crowdleague/actions/device/check_platform.dart';
```
Comment (tamari-gray) [inline test/unit/middleware/check_platform_middleware_test.dart:1]: this is refactoring this test to use our new approach to testing middleware, rather then checking that the state is updated, we are checking that our mock store has dispatched what we want it to!

**Review by tamari-gray — COMMENTED** [review]
ready for review 😄

**Comment by tamari-gray** [comment]
71 passing tests!!

File: test/mocks/redux/redux_store_mocks.dart [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+class MockStore extends Mock implements Store<AppState> {}
+
+class DispatchVerifyingStore implements Store<AppState> {
+  @override
+  var reducer;
+
+  final AppState _state;
+  List<ReduxAction> dispatchedActions = <ReduxAction>[];
```
[the remark this answers is not in the document]
Comment (nickmeinhold) [reply test/mocks/redux/redux_store_mocks.dart:22]: haha love it!

File: test/unit/middleware/auth/observe_auth_state_middleware_test.dart [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import 'package:built_collection/built_collection.dart';
```
Comment (nickmeinhold) [inline test/unit/middleware/auth/observe_auth_state_middleware_test.dart:1]: great test!

File: test/unit/middleware/auth/sign_in_with_apple_middleware_test.dart [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import 'package:crowdleague/actions/auth/sign_in_with_apple.dart';
```
Comment (nickmeinhold) [inline test/unit/middleware/auth/sign_in_with_apple_middleware_test.dart:1]: Love this test - go team!

File: test/unit/middleware/auth/sign_in_with_email_middleware_test.dart [hunk 6]
```diff
@@ -0,0 +1,70 @@
+import 'package:crowdleague/actions/auth/sign_in_with_email.dart';
```
Comment (nickmeinhold) [inline test/unit/middleware/auth/sign_in_with_email_middleware_test.dart:1]: This is brilliant!  It's not just some awesome tests but it's also providing some sweet documentation!

File: test/unit/middleware/check_platform_middleware_test.dart (hunk 2 above)
Comment (nickmeinhold) [reply test/unit/middleware/check_platform_middleware_test.dart:1]: Thanks for the clean up - looks great!

**Review by nickmeinhold — CHANGES_REQUESTED** [review]
Amazing work mate!  Just the question around awaiting the calls to dispatch.  Kick ass PR - thanks for the awesome work!

File: lib/services/auth_service.dart [hunk 7]
```diff
@@ -190,7 +191,7 @@ class AuthService {
       await FirebaseAuth.instance.signOut();
       await _googleSignIn.signOut();
       // TODO: add sign out for sign in with apple provider
-      // see Issue #6 https://github.com/nickmeinhold/crowdleague_public/issues/6
+      // see Issue #232 https://github.com/crowdleague/crowdleague/issues/232
```
[the remark this answers is not in the document]
Comment (tamari-gray) [reply lib/services/auth_service.dart:194]: sweet no worries!

File: test/unit/middleware/auth/observe_auth_state_middleware_test.dart (hunk 4 above)
Comment (tamari-gray) [reply test/unit/middleware/auth/observe_auth_state_middleware_test.dart:1]: thank you 🙇 🙇

File: test/unit/middleware/auth/sign_in_with_email_middleware_test.dart (hunk 6 above)
Comment (tamari-gray) [reply test/unit/middleware/auth/sign_in_with_email_middleware_test.dart:1]: true!

File: test/unit/middleware/auth/sign_in_with_apple_middleware_test.dart (hunk 5 above)
Comment (tamari-gray) [reply test/unit/middleware/auth/sign_in_with_apple_middleware_test.dart:1]: 🥇

**Review by nickmeinhold — APPROVED** [review]
Woohoo! Thanks heaps!!!

**Pull request merged**